src/formats/base_32.cpp

```diff
@@ -122,8 +122,8 @@ data_chunk checksum(const base32& value)
     data_chunk checksum(checksum_size);
     const auto modified = polymod(expanded) ^ 1;
 
-    for (size_t index = 0; index < checksum.size(); ++index)
-        checksum[index] = (modified >> (5 * (5 - index))) & 31;
+    for (size_t index = 0; index < checksum_size; ++index)
```

Comment:
Use `auto` (no need for more than int here).

Reply:
After considering, I left this (and a different instance, below) as-is to avoid compiler warnings:

```
src/formats/base_32.cpp: In function ‘uint32_t libbitcoin::system::polymod(const data_chunk&)’:
src/formats/base_32.cpp:102:36: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
         for (auto index = 0; index < bit_group_size; ++index)
                              ~~~~~~^~~~~~~~~~~~~~~~
src/formats/base_32.cpp: In function ‘libbitcoin::system::data_chunk libbitcoin::system::checksum(const libbitcoin::system::base32&)’:
src/formats/base_32.cpp:127:32: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
     for (auto index = 0; index < checksum_size; ++index)
```
Alternatively, could use int for the constants ...